### PR TITLE
feat(DatePicker): support confirm and getSelectedDate method

### PR DESCRIPTION
**Pull request by bluesky335**
issue：https://github.com/youzan/vant/issues/12711
为 DatePicker 组件添加了 confirm 方法
为 DatePicker 组件添加了 getSelectedDate 方法

### CODE REVIEW

File: packages/vant/src/date-picker/DatePicker.tsx [hunk 1]
```diff
@@ -121,6 +134,14 @@ export default defineComponent({
       return genOptions(minDate, maxDate, 'day', props.formatter, props.filter);
     };
 
+    const confirm = () => {
+      return pickerRef.value?.confirm();
+    };
+
+    const getSelectedDate = (): Array<string> => {
```
Comment (chouchouji) [inline packages/vant/src/date-picker/DatePicker.tsx:141]: The type of **getSelectedDate** can be inferred by ts, maybe no need to declare the return type.

File: packages/vant/src/date-picker/README.md [hunk 2]
```diff
@@ -204,10 +204,34 @@ export default {
 | columns-top | Custom content above columns | - |
 | columns-bottom | Custom content below columns | - |
 
+### Methods
+
+Use [ref](https://vuejs.org/guide/essentials/template-refs.html) to get Picker instance and call instance methods.
+
+| Name | Description | Attribute | Return value |
+| --- | --- | --- | --- |
+| confirm | Stop scrolling and emit confirm event | - | - |
```
[the remark this answers is not in the document]
Comment (chenjiahan) [reply packages/vant/src/date-picker/README.md:213]: As far as I know confirm method has no return value..

File: packages/vant/src/date-picker/DatePicker.tsx (hunk 1 above)
Comment (chenjiahan) [inline packages/vant/src/date-picker/DatePicker.tsx:141]:
```suggestion
    const getSelectedDate = () => {
```

File: packages/vant/src/date-picker/DatePicker.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {
   watch,
   computed,
   defineComponent,
+  ComponentPublicInstance,
```
Comment (chenjiahan) [inline packages/vant/src/date-picker/DatePicker.tsx:6]:
```suggestion
  type ComponentPublicInstance,
```

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**